### PR TITLE
fix(console): SAML authn form and wizard hotfix port

### PR DESCRIPTION
**Pull request by yasinmiran**
### Purpose
Changelist:
- Updated SAML & OIDC IdP name validation logic.
- Make SAML signing optional (Port of https://github.com/wso2/identity-apps/commit/d6d25cc76ad770f949cd905c47e142ab93b96f29)
- Adds a failsafe default value for SAML IdP file configuration.

### Related Issues
- Issue `#1` or (None)

### Checklist
- [ ] e2e cypress tests locally verified.
- [x] Manual test round performed and verified.
- [x] UX/UI review done on the final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Related PRs
- Related PR `#1` or (None)

### Security checks
- [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx [hunk 1]
```diff
@@ -166,7 +174,7 @@ export const SamlAuthenticatorSettingsForm: FunctionComponent<SamlSettingsFormPr
         setIsAlgorithmsEnabled(ifEitherOneOfThemIsChecked);
     }, [ isLogoutReqSigned, isAuthnReqSigned ]);
 
-    const onFormSubmit = (values: { [ key: string ]: any }): void => {
+    const onFormSubmit = (values: { [key: string]: any }): void => {
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx:177]:
```suggestion
    const onFormSubmit = (values: { [ key: string ]: any }): void => {
```

File: apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts [hunk 2]
```diff
@@ -258,7 +266,7 @@ export const booleanSentAsAStringValue = (value: any): boolean => {
 };
 
 export type MinMaxLength = { max: number; min: number };
-export type FormErrors = { [ key: string ]: string };
+export type FormErrors = { [key: string]: string };
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts:269]:
```suggestion
export type FormErrors = { [ key: string ]: string };
```

File: apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts [hunk 3]
```diff
@@ -276,7 +284,7 @@ export const IDENTITY_PROVIDER_NAME_LENGTH: MinMaxLength = { max: 120, min: 3 };
  * @param errors {FormErrors}
  */
 export const ifFieldsHave = (errors: FormErrors): boolean => {
-    return !Object.keys(errors).every((k) => !errors[ k ]);
+    return !Object.keys(errors).every((k) => !errors[k]);
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts:287]:
```suggestion
    return !Object.keys(errors).every((k) => !errors[ k ]);
```

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx [hunk 4]
```diff
@@ -388,6 +388,12 @@ export const EnterpriseIDPCreateWizard: FC<EnterpriseIDPCreateWizardProps> = (
                 errors.name = t("console:develop.features.authenticationProvider." +
                     "forms.generalDetails.name.validations.duplicate");
             }
+            // We have the inputType "identifier" but however it has no
+            // effect because validate function is overridden by this function.
+            // So, we need re-check this manually.
+            if (!FormValidation.identifier(values.name)) {
+                errors.name = "Please enter a valid name";
```
Comment (thivi) [inline apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx:395]: Ideally, this should be localized, right?

**Review by thivi — APPROVED** [review]

File: apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx (hunk 1 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/forms/authenticators/saml-authenticator-form.tsx]: Addressed with e768e73

File: apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts (hunk 3 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts]: Addressed with e768e73

File: apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts (hunk 2 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/utils/saml-idp-utils.ts]: Addressed with e768e73

File: apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx (hunk 4 above)
Comment (yasinmiran) [reply apps/console/src/features/identity-providers/components/wizards/enterprise-idp-create-wizard.tsx]: Addressed with 8c6e4d1

**Pull request merged**